askama_derive/src/generator.rs

```diff
@@ -1037,16 +1040,16 @@ impl<'a> Generator<'a> {
             .config
             .find_template(path, Some(&self.input.path))?;
 
-        enum CowNodes<'a> {
-            Owned(Parsed),
-            Borrowed(&'a [Node<'a>]),
-        }
-
         let nodes = match self.contexts.get(path.as_path()) {
-            Some(ctx) => CowNodes::Borrowed(&ctx.nodes),
+            Some(ctx) => ctx.nodes,
             None => {
-                let src = get_template_source(&path)?;
-                CowNodes::Owned(Parsed::new(src, self.input.syntax)?)
+                if !self.includes.contains_key(path.as_path()) {
```

Comment:
I tried to do this with an `Entry` to avoid double-hashing, but couldn't quite get that to work due to the limited lifetime of the `Entry` value. I think the overhead of extra hashing is probably not worth it.

Reply:
This should work:

```rust
None => match self.includes.entry(path) {
    Entry::Occupied(entry) => entry.into_mut().nodes(),
    Entry::Vacant(entry) => {
        let src = get_template_source(entry.key())?;
        entry.insert(Parsed::new(src, self.input.syntax)?).nodes()
    }
}
```

Then the "Make sure the compiler understands that the generated code depends on the template file." block needs to come before `let nodes = …`.